### PR TITLE
Initial port to Swift 4

### PR DESCRIPTION
**Pull request by tcldr**
What's in this pull request?
============================

Initial migration of library to Swift 4. (Minor changes but lib required by Swiftz which has more extensive changes)

Why merge this pull request?
============================

Migrates library to latest version of Swift. All tests passing with the Swift 4 version of SwiftCheck that I submitted as a PR earlier.

What's worth discussing about this pull request?
================================================

This PR focuses on getting the build compiling without warning and passing all tests. I will have missed anything not picked up by these! There may be some wider changes you have in mind to take advantage of Swift 4. 

What downsides are there to merging this pull request?
======================================================

Uses temporary refs to my personal SwiftCheck forks with Swift 4 migration. Will need updating to point to official SwiftCheck repos.

### CODE REVIEW

**Comment by bkase** [comment]
Alright, I merged the SwiftCheck PR -- so now we can point to that and then we can merge this one!

**Comment by tcldr** [comment]
Nice one. I've updated Operadics to SPM v4 too. Once that's in we can cascade the rest up!

File: Package.resolved [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (CodaFi) [inline Package.resolved:1]: This file should be ignored

File: Sources/Operators.swift [hunk 2]
```diff
@@ -0,0 +1,236 @@
+//
```
Comment (CodaFi) [inline Sources/Operators.swift]: We should take a dependency on Operadics, not include it directly.

**Comment by CodaFi** [comment]
Xcode 9 redid the file paths so they match the project structure.  This is undesirable given that I would like to support just a package build in the future.  Could you please try to keep everything in their original positions (modulo moving things into the target directory)?

File: Cartfile.private [hunk 3]
```diff
@@ -1,3 +1,3 @@
-github "typelift/Operadics" 
-github "typelift/SwiftCheck"
+github "typelift/Operadics" "swift-develop"
```
Comment (CodaFi) [inline Cartfile.private]: You may remove these.  I've just pushed Xcode 9 updates to SwiftCheck and Operadics

File: Package.resolved (hunk 1 above)
Comment (tcldr) [reply Package.resolved:1]: Sure. Added to `.gitignore` along with the Cartfile resolved. My understanding was that these files *should* be committed however to ensure that those using the project are using the last known working versions of any dependencies.

File: Sources/Operators.swift (hunk 2 above)
Comment (tcldr) [reply Sources/Operators.swift]: Good spot. Not sure how that got in there.

File: Package.resolved (hunk 1 above)
Comment (CodaFi) [reply Package.resolved:1]: Sure, but Operadics is essentially frozen, and the other two are test-only dependencies.  It doesn’t matter which version our own internal repos are built against, it won’t affect our clients.

**Comment by CodaFi** [comment]
Resolved by the merge of #68

**Pull request closed without merging**